lib/tvdbapiv2/api_client.py

```diff
@@ -67,52 +61,36 @@ class ApiClient(object):
     :param header_name: a header to pass when making calls to the API.
     :param header_value: a header value to pass when making calls to the API.
     """
-    def __init__(self, host=None, cookie=None, session=None, user_agent=None, api_key=None):
+    def __init__(self, host=None, cookie=None, session=None, api_key=None, token=None):
 
         """
         Constructor of the class.
         """
         self.session = session or requests.session()
-        if api_key:
-            self.session.auth = TVDBAuth(api_key)
+        if not api_key and not token:
+            raise MissingAuthentication('Please provide an api_key or token to authenticate.')
+
+        try:
+            self.session.auth = TVDBAuth(api_key=api_key, token=token)
```

Comment:
Lazy does not mean it won't try to auth if the network is down. It means it won't try to auth until the session is used. If the session keeps getting used while the connection is down, it will still keep trying to auth.